src/frontend/src/pages/AdminToolsPage/TeamsTools.tsx

```diff
@@ -0,0 +1,72 @@
+import { Box, Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@mui/material';
+import PageBlock from '../../layouts/PageBlock';
+import { NERButton } from '../../components/NERButton';
+import { useAllTeams } from '../../hooks/teams.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { fullNamePipe } from '../../utils/pipes';
+import CreateTeamModal from './CreateTeamModal';
+import { useState } from 'react';
+
+const TeamsTools = () => {
+  const [showCreateModal, setShowCreateModal] = useState(false);
+  const { data: allTeams, isLoading, isError, error } = useAllTeams();
+
+  if (!allTeams || isLoading) return <LoadingIndicator />;
+
+  if (isError) {
+    return <ErrorPage message={error.message} />;
+  }
+
+  const teamTableRows = allTeams.map((team) => (
+    <TableRow>
+      <TableCell sx={{ border: '2px solid black' }}>{team.teamName}</TableCell>
+      <TableCell sx={{ border: '2px solid black' }}>{fullNamePipe(team.head)}</TableCell>
+      <TableCell sx={{ border: '2px solid black' }}>{team.members.length}</TableCell>
+    </TableRow>
+  ));
+
+  return (
+    <PageBlock title="Team Management">
+      <CreateTeamModal showModal={showCreateModal} handleClose={() => setShowCreateModal(false)} />
+      <TableContainer component={Paper}>
+        <Table>
+          <TableHead>
+            <TableCell
```

Comment:
I feel like i said we should make this a custom component

Reply:
sure